src/reanimated2/shareableCache.ts

```diff
@@ -12,4 +23,22 @@ There we use Object.entries to iterate over the keys which throws an error on ac
 For convenience we move this cache to a separate file so it doesn't scare us with red squiggles.
 */
 
-export const shareableCache = new WeakMap<object, ShareableRef | symbol>();
+const cache = SHOULD_BE_USE_WEB
+  ? null
+  : new WeakMap<object, ShareableRef | symbol>();
+
+export const shareableCache = cache
+  ? {
+      set(shareable: object, shareableRef?: ShareableRef): void {
+        cache.set(shareable, shareableRef || shareableFlag);
+      },
+      get: cache.get,
+    }
+  : {
+      set() {
+        // NOOP
+      },
+      get() {
+        return null;
```

Comment:
Maybe let's throw an error here that says "`shareableCache.get` should not be called on web"? Or, even better, we could add some helper error function that would be called here.

Reply:
What do you mean by 'helper error function'?

Reply:
Some function that we call from here and other places that shouldn't be called on web that throws an error that this function should not be called on web. Something like `__builtin_trap()`.

Reply:
If you want to do it in other places that seems like a separate PR. In this one I didn't want to change the behavior, since currently on web `shareableCache` is always empty.